src/python_bindings/py_data_stats.cpp

```diff
@@ -0,0 +1,162 @@
+#include "py_data_stats.h"
+
+#include "statistics/data_stats.h"
+
+namespace python_bindings {
+
+namespace py = pybind11;
+namespace mo = model;
+
+size_t PyDataStats::GetNumberOfValues(size_t index) const {
+    return GetAlgorithm<algos::DataStats>(algorithm_).NumberOfValues(index);
```

Comment:
You should make a function template for this and use it everywhere applicable.

Reply:
Nevermind, it doesn't look good in this case. I was thinking of something like this (doesn't work for all cases):
```cpp
template <typename T, typename... Args>
auto GetDirectStatValue(std::unique_ptr<algos::Algorithm> const& data_stats,
                        T(algos::DataStats::*method)(Args...) const, Args... args) {
    return (GetAlgorithm<algos::DataStats>(data_stats).*method)(args...);
}
```